tests/test_legacy.F90

```diff
@@ -0,0 +1,124 @@
+! (C) Copyright 2022- ECMWF.
+! (C) Copyright 2022- Meteo-France.
+!
+! This software is licensed under the terms of the Apache Licence Version 2.0
+! which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+! In applying this licence, ECMWF does not waive the privileges and immunities
+! granted to it by virtue of its status as an intergovernmental organisation
+! nor does it submit to any jurisdiction.
+
+PROGRAM TEST_LEGACY
+
+USE FIELD_MODULE
+USE FIELD_FACTORY_MODULE
+USE FIELD_ACCESS_MODULE
+USE FIELD_ABORT_MODULE
+USE FIELD_CONSTANTS_MODULE
+USE FIELD_UTIL_MODULE, ONLY : LEGACY, LEGACY_FIELD_3RD
+
+USE PARKIND1
+
+IMPLICIT NONE
+
+CLASS(FIELD_3RB), POINTER :: FU, FV, FT
+
+REAL(KIND=JPRB), ALLOCATABLE :: GMV (:,:,:,:)
+
+REAL(KIND=JPRB), POINTER :: U (:,:,:), V (:,:,:), T (:,:,:)
+
+INTEGER, PARAMETER :: NFLEVG = 5, NPROMA = 32, NGPBLKS = 4, NDIM = 3
+INTEGER, PARAMETER :: IU = 1, IV = 2, IT = 3
+
+INTEGER*8 :: IADDRL, IADDRU
+
+INTEGER :: JLON, JLEV, JBLK
+INTEGER :: IERR
+
+ALLOCATE (GMV (NPROMA, NFLEVG, NDIM, NGPBLKS))
+
+GMV = 0._JPRB
+
+CALL FIELD_NEW (FU, DATA=GMV (:,:,IU,:))
+CALL FIELD_NEW (FV, DATA=GMV (:,:,IV,:))
+CALL FIELD_NEW (FT, DATA=GMV (:,:,IT,:))
+
+! All fields on the device
+
+U => GET_DEVICE_DATA_RDWR (FU)
+V => GET_DEVICE_DATA_RDONLY (FV)
+T => GET_DEVICE_DATA_RDONLY (FT)
+
+!$acc kernels present (U)
+U = 1._JPRB
+!$acc end kernels
+
+!$acc enter data create (GMV)
+!$acc update device (GMV)
```

Comment:
[no action] Since GMV has been allocated by the time we get here, we can simply do `!$acc enter data copyin`.